ngraph/test/frontend/shared/src/op_fuzzy.cpp

```diff
@@ -159,5 +158,5 @@ TEST_P(FrontEndFuzzyOpTest, testOpFuzzy)
     ASSERT_NE(function, nullptr);
 
     // run
-    runConvertedModel(function, m_modelFile);
+    runConvertedModel<test::INTERPRETER_Engine>(function, m_modelFile);
```

Comment:
Why do we need to call inference here instead of just comparing nGraph function with reference?

Reply:
Hi it's because, in our Fuzzy Ops Test, we need to compute a result from `ngraph reference` and compare that result with the output value saved in `xxx.cpy` of the PDPD model.

Reply:
Why cannot we use e2e tests to compare results?
Because when you call reference implementation you introduce the new component in your tests.
I mean that in this case we can have a bug in the reference implementation, it is not a part of PDPD frontend.
If we compare functions we use only functionality of frontend and avoid additional code in our test pipeline

Reply:
@ilyachur agree with you, no e2e test to avoid introducing new components sounds make sense.

Problem is, how do I get the reference nG function?
In some case of decomposition, e.g. yolo-box, I have to run inference and compare to PDPD result to know whether the decomposition is right. There is nowhere I can get its reference.

Reply:
@ceciliapeng2011 Maybe I am wrong, but from my point of view it is really strange use inference in order to get nGraph function. Because reference implementations can contain bug and in this case for right results you also can get incorrect reference function (if reference implementation has a bug).

Moreover Model Optimizer converts PDPD to IR and has tests without using inference to compare that function was converted correctly. I think MO uses another idea to compare correctness of conversion.

Reply:
@nosovmik Could you share us on how MO tests and compare correctness of conversion?

@ilyachur 
This op Fuzzy test is to validate the conversion from PDPD to nG is correct. That means, given the same inputs, PDPD frontend 1:n mapping/decomposition should give out the same outputs exactly as PDPD. 

If not, possibly the conversion itself is wrong, or something wrong in inference. 
If it is conversion wrong, we have to modify the mapping/decomposition. So there is no reference nG function we can use to compare.
If it is wrong inference, the problem would appear in some kind of e2e test sooner or later. Our Op conversion depends on the correct nG reference implementation. The purpose of op conversion should not just convert to kind of nG function, but to make sure it has exact same result of PDPD. 
From this perspective of view, **our op Fuzzy test should be e2e test**. 

As to your question about involving extra dependency to ngraph/test/frontend, how about let's move it out to somewhere else under ngraph/test? For example, a place like ngraph/test/paddlepaddle, there already been one onnx there.

Reply:
I am thinking from the perspective that in future we will have some set of frontends (not only ONNX and PDPD), moreover customers will be able to register own frontends.
And it won't be a good solution to have such tests with inference for each framework. Will be better to have tests which creates different subgraph and compare it with reference graphs. And also have separate tests which check IE plugins and reference implementations on operations from these subgraphs. In other case we make a double work because check execution in SLT and here in the frontend tests.

About extra dependency I said because I see each frontend as a separate component, if customer develop own frontend he should test only own components and shouldn't test our reference implementations 😄 

If we are talking about e2e tests we have a separate infrastructure which validate e2e scenario. It takes real models and run it with different plugins.